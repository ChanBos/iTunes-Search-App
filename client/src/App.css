/* Set the size (height x width), text alignment and background and text colour of the application. */
#App,
#root,
body {
  height: auto;
  width: 100%;
  text-align: center;
  background-color: #000000 !important;
  color: #ffffff;
}

/* --- Header --- */
/* Set for the header to display as a grid, set the frames, the position to absolute and aligned it to the top-left corner and set the z-index 
to 1 so that the header appears over the search container. Also set the size (height x width), the padding, the background colour and the font 
size and colour. */
#header {
  display: grid;
  grid-template-columns: 1fr 1fr;
  position: absolute;
  left: 0px;
  top: 0px;
  height: auto;
  width: 100%;
  padding: 10px;
  font-size: calc(10px + 2vmin);
  background-color: transparent;
  color: #ffffff;
  z-index: 1 !important;
}

/* Set for the logo's column to take up the entire available space, taking into account the navigation bar (left margin set to auto). */
#header:nth-child(1) {
  grid-column-start: 2;
}
#header:nth-child(2) {
  margin-left: auto;
}

/* Adjusted the size (height x width) and set for the logo to float to the left. */
#logo {
  height: 80px;
  width: auto;
  float: left;
}

/* --- Navigation bar --- */
/* Set for the navigation container to float to the right. */
.navbar {
  float: right;
}

/* Set the width. top padding and font colour and size of the navigation links. */
.nav-link {
  width: 160px;
  padding-top: 17px !important;
  font-size: 1.3rem;
  color: #ffffff !important;
}

/* --- Search --- */
/* Set the position of the search container to relative, altered the left position and transformed it to be centered on the x (horizontal 
axis. Also set the size (height x width) and the margins. */
#searchcontainer {
  position: relative;
  left: 50%;
  transform: translateX(-50%);
  height: auto;
  width: 1250px;
  margin: -20px 0px 0px;
}

/* Set the search contents' size (height x width), margins, top padding, added a background image and set the background colour. */
#searchcontent {
  height: 600px;
  width: auto;
  margin: auto;
  padding-top: 295px;
  background-image: url("images/background.png");
  background-color: #000000;
}

/* Set the search button, dropdown and input section's height. Also added a border (styled, sized, coloured and radius set). */
#searchinput,
#searchbutton {
  height: 40px;
  border: 1px solid #000000;
  border-radius: 5px;
}

/* Set for the search input container to display as flex and the direction to row. Also set the size (height x width), the margins, the 
background colour and the padding of the search input area. */
#searchinput {
  display: flex;
  flex-direction: row;
  height: 40px;
  width: 600px;
  margin: 0px auto 15px;
  padding: 8px;
  background-color: #ffffffb7;
}

/* Set for the input area to take up all of the available space by setting flex-grow, set the left-side padding and removed the border and 
background colour. */
input {
  background-color: transparent;
  border: none;
  flex-grow: 2;
  padding-left: 5px;
}

/* Removing the input focus blue box by setting outline to none.*/
input:focus {
  outline: none;
}

/* Set the font weight, colour and opacity of the input section's placeholder text. */
::placeholder {
  font-weight: 650;
  color: #272727;
  opacity: 1;
}

/* Set the search button's size (height x width), top margin and removed the border and background colour. */
#searchbutton {
  height: auto;
  width: auto;
  margin-top: -15px;
  background-color: transparent;
  border: none;
}

/* --- Dropdown Menu ---*/
/* Set the width and the background colour of the dropdown menu button and options. */
.dropdown-menu,
.btn-md {
  width: 125px;
  background-color: #0000008e !important;
}

/* Set the font size and colour of the dropdown options. */
.dropdown-item {
  font-size: 0.8rem;
  color: #ffffff !important;
}

/* Set the highlight background colour for when the options are hovered over. */
.dropdown-item:hover {
  background-color: #5120569d !important;
}

/* --- Cards --- */
/* Set the size (height x width), padding, background colour and opacity of the cards. Added a gradient overlay (styled, coloured, percentages
set and degree stipulated. Also added a border (styled, coloured, sized and border radius adjusted. */
.card {
  height: 470px;
  width: 100%;
  padding: 15px;
  background-color: #000000;
  background-image: linear-gradient(315deg, #414141 0%, #000000 74%);
  border: 1px solid #000000;
  border-radius: 10px !important;
  opacity: 0.8;
}

/* Set the card footer's padding to 0. */
.card-footer {
  padding: 0px !important;
}

/* Set the audio preview button's margins and padding. */
#previewbutton {
  margin: auto auto 10px;
  padding: 3px 0px 0px;
}

/* Set the audio's size (height x width). */
audio {
  height: 35px;
  width: 235px;
}

/* Set the cards' button background colour and added a border (styled, sized and coloured). */
.cardfuncbutton {
  background-color: #303030 !important;
  border: 1px solid #414040 !important;
}

/* Set the "Add Fav" button's font size and padding. */
#addfavbutton {
  font-size: 0.91rem;
  padding: 7.5px;
}

/* Set for the "Add Fav" button's border colour to change once hovered over. */
#addfavbutton:hover {
  border: 1px solid #ff2589 !important;
}

/* Set for the "More Info" button's border colour to change once hovered over. */
.infobutton:hover {
  border: 1px solid #237acd !important;
}

/* Set for the "Delete" button's border colour to change once hovered over. */
#deletebutton:hover {
  border: 1px solid #ff5e50 !important;
}

/* --- Results --- */
/* Set the top margin of the result container. */
#resultcontainer {
  margin-top: 30px !important;
}

/* --- Favourites --- */
/* Set the position of the favourite container to relative, altered the left position and transformed it to be centered on the x (horizontal 
axis. Also set the size (height x width) and the margins. */
#favouritecontainer {
  position: relative;
  left: 50%;
  transform: translateX(-50%);
  height: auto;
  width: 1250px;
  margin: 0px;
}

/* Set the top padding and the bottom margin of the favourite's heading. */
h3 {
  padding-top: 95px !important;
  margin-bottom: 20px !important;
}

/* Set the alert modal's text colour of the title. */
.swal2-title {
  color: #ffffff !important;
}

/* --- Footer --- */
/* Set the size (height x width), the background colour and padding of the footer. */
#footer {
  height: 55px;
  width: 100%;
  padding: 10px;
  background-color: transparent !important;
}

/* Set the font colour and the margins of the footer's header text. */
h6 {
  margin: 0px auto !important;
  color: #ffffff;
}

/* Set the margins, the font colour and the text decoration to none for the footer link. */
a {
  margin: auto 6px;
  color: #ffffff !important;
  text-decoration: none !important;
}

/* --- Icons --- */
/* Set the font size of the fa-search search icon. */
.fa-search {
  font-size: 1.15rem;
  margin-left: 8px;
}

/* Set for the font colour and the cursor to change once the fa-search search icon is hovered over. */
.fa-search:hover {
  color: #e244b0;
  cursor: pointer;
}

/* Set the margins of the icons. */
.fa-home,
.fa-star,
.fa-apple,
.fa-trash-alt {
  margin-right: 5px;
}

/* Overridden the bottom and right margins and font size of the "Add Fav" button's star icon. */
#resultcontainer button .fa-star {
  margin-bottom: 1px !important;
  margin-right: 4px;
  font-size: 0.8rem;
}

/* Set the font colour of the footer's heart icon. */
.fa-heart {
  color: #512056;
}

/* Set the bottom margin, the font size of the "Scroll to Top" icon and for the cursor to change once the icon is hovered over. */
.fa-angle-double-up {
  margin-bottom: 12px !important;
  font-size: 1.75rem;
  cursor: pointer;
}

/* Set the font colour of the footer's "Scroll to Top" icon. */
.fa-angle-double-up:hover {
  color: #512056;
}
