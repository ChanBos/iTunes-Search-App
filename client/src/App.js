// Imported React libraries.
import React from "react";
// Imported Route, Switch and Browser Router from React Router Dom.
import { Route, Switch, BrowserRouter } from "react-router-dom";
// Imported stylesheet.
import "./App.css";
// Imported components.
import Favourites from "./components/favourites.js";
import Landing from "./components/landing.js";
import Header from "./components/header.js";
import Footer from "./components/footer.js";

/**
 * Created the App function and returning components to be exported to Index.js.
 * Used "BrowserRouter", "Switch and "Route" to show components to manage the navigation of the application.
 * Added the React Bootstrap link.
 * @returns Application that returns all of the components.
 */

const App = () => {
  return (
    <div id="App">
      <BrowserRouter>
        <div>
          <link
            rel="stylesheet"
            href="https://stackpath.bootstrapcdn.com/bootstrap/4.4.1/css/bootstrap.min.css"
            integrity="sha384-Vkoo8x4CGsO3+Hhxv8T/Q5PaXtkKtu6ug5TOeNV6gBiFeWPGFN9MuhOf23Q9Ifjh"
            crossOrigin="anonymous"
          ></link>
          <Header />
          <Switch>
            <Route path="/" component={Landing} exact={true} />
            <Route path="/favourites" component={Favourites} />
          </Switch>
          <Footer />
        </div>
      </BrowserRouter>
    </div>
  );
};

// Exported App.js to index.js.
export default App;
