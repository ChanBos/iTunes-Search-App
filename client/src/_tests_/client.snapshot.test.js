// Imported React libraries.
import React from "react";
// Imported App.js for testing.
import App from "../App";
// Imported react-test-renderer for rendering snapshots.
import renderer from "react-test-renderer";

/**
 * Created and described the test being performed. Constructed a tree variable, storing the JSON that (represents the
 * DOM tree that is created when the component we are testing is rendered). Created the App component via the create() method and serialized
 * into JSON objects.
 */

test("Renders without crashing?", () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});
