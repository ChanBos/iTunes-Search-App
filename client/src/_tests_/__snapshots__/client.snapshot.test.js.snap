// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders without crashing? 1`] = `
<div
  id="App"
>
  <div>
    <link
      crossOrigin="anonymous"
      href="https://stackpath.bootstrapcdn.com/bootstrap/4.4.1/css/bootstrap.min.css"
      integrity="sha384-Vkoo8x4CGsO3+Hhxv8T/Q5PaXtkKtu6ug5TOeNV6gBiFeWPGFN9MuhOf23Q9Ifjh"
      rel="stylesheet"
    />
    <div>
      <div
        className="container-fluid"
        id="header"
      >
        <div
          id="logocontainer"
        >
          <img
            alt="Logo"
            id="logo"
            src="./images/logo.png"
          />
        </div>
        <div
          id="navcontainer"
        >
          <nav
            className="navbar navbar-expand navbar-light"
          >
            <div
              className="container"
            >
              <div
                className="row"
              >
                <div
                  className="col"
                >
                  <div
                    className="navbar-nav"
                    onKeyDown={[Function]}
                  >
                    <div
                      className="nav-item"
                    >
                      <a
                        className="nav-link"
                        data-rb-event-key="/favourites"
                        href="/favourites"
                        onClick={[Function]}
                        onKeyDown={[Function]}
                      >
                        <svg
                          aria-labelledby="svg-inline--fa-title-fjjBUR7LNPeB"
                          className="svg-inline--fa fa-star fa-w-18 "
                          data-icon="star"
                          data-prefix="fas"
                          role="img"
                          style={Object {}}
                          viewBox="0 0 576 512"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <title
                            id="svg-inline--fa-title-fjjBUR7LNPeB"
                            style={Object {}}
                          >
                            Favourites
                          </title>
                          <path
                            d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
                            fill="currentColor"
                            style={Object {}}
                          />
                        </svg>
                        FAVOURITES
                      </a>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </nav>
        </div>
      </div>
    </div>
    <div
      className="container"
    >
      <div
        id="searchcontainer"
      >
        <div
          id="searchcontent"
        >
          <div
            id="searchinput"
          >
            <input
              name="name"
              onChange={[Function]}
              placeholder="Search..."
              type="text"
            />
            <a
              href="/search"
              onClick={[Function]}
            >
              <button
                id="searchbutton"
                onClick={[Function]}
                type="submit"
              >
                <svg
                  aria-labelledby="svg-inline--fa-title-riFSJdmEuNcq"
                  className="svg-inline--fa fa-search fa-w-16 "
                  data-icon="search"
                  data-prefix="fas"
                  role="img"
                  style={Object {}}
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <title
                    id="svg-inline--fa-title-riFSJdmEuNcq"
                    style={Object {}}
                  >
                    Search
                  </title>
                  <path
                    d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
              </button>
            </a>
          </div>
          <div
            className="dropright btn-group"
            role="group"
          >
            <button
              aria-expanded={false}
              aria-haspopup={true}
              className="dropdown-toggle btn btn-info btn-md"
              disabled={false}
              id="dropdown-basic"
              onClick={[Function]}
              type="button"
            >
              CATEGORIES
            </button>
          </div>
        </div>
        <div
          className="search-page"
        >
          <div
            className="container-fluid"
          >
            <div
              className="row row-cols-md-4"
            >
              <div />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      id="footer"
    >
      <div
        className="container"
      >
        <div
          className="row"
        >
          <h6>
            PROUDLY CREATED WITH
             
            <svg
              aria-labelledby="svg-inline--fa-title-lbd7pgDwMcvN"
              className="svg-inline--fa fa-heart fa-w-16 "
              data-icon="heart"
              data-prefix="fas"
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title
                id="svg-inline--fa-title-lbd7pgDwMcvN"
                style={Object {}}
              >
                Favourites
              </title>
              <path
                d="M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
             USING THE
            <a
              href="https://affiliate.itunes.apple.com/resources/documentation/itunes-store-web-service-search-api/"
              rel="noreferrer"
              target="_blank"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-apple fa-w-12 "
                data-icon="apple"
                data-prefix="fab"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 384 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M318.7 268.7c-.2-36.7 16.4-64.4 50-84.8-18.8-26.9-47.2-41.7-84.7-44.6-35.5-2.8-74.3 20.7-88.5 20.7-15 0-49.4-19.7-76.4-19.7C63.3 141.2 4 184.8 4 273.5q0 39.3 14.4 81.2c12.8 36.7 59 126.7 107.2 125.2 25.2-.6 43-17.9 75.8-17.9 31.8 0 48.3 17.9 76.4 17.9 48.6-.7 90.4-82.5 102.6-119.3-65.2-30.7-61.7-90-61.7-91.9zm-56.6-164.2c27.3-32.4 24.8-61.9 24-72.5-24.1 1.4-52 16.4-67.9 34.9-17.5 19.8-27.8 44.3-25.6 71.9 26.1 2 49.9-11.4 69.5-34.3z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
              iTunes
            </a>
            SEARCH API.
          </h6>
          <div
            id="scrollcontainer"
          >
            <svg
              aria-labelledby="svg-inline--fa-title-mJuBYOrpsGxc"
              className="svg-inline--fa fa-angle-double-up fa-w-10 "
              data-icon="angle-double-up"
              data-prefix="fas"
              onClick={[Function]}
              role="img"
              style={
                Object {
                  "display": "none",
                }
              }
              viewBox="0 0 320 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title
                id="svg-inline--fa-title-mJuBYOrpsGxc"
                style={Object {}}
              >
                Scroll to Top
              </title>
              <path
                d="M177 255.7l136 136c9.4 9.4 9.4 24.6 0 33.9l-22.6 22.6c-9.4 9.4-24.6 9.4-33.9 0L160 351.9l-96.4 96.4c-9.4 9.4-24.6 9.4-33.9 0L7 425.7c-9.4-9.4-9.4-24.6 0-33.9l136-136c9.4-9.5 24.6-9.5 34-.1zm-34-192L7 199.7c-9.4 9.4-9.4 24.6 0 33.9l22.6 22.6c9.4 9.4 24.6 9.4 33.9 0l96.4-96.4 96.4 96.4c9.4 9.4 24.6 9.4 33.9 0l22.6-22.6c9.4-9.4 9.4-24.6 0-33.9l-136-136c-9.2-9.4-24.4-9.4-33.8 0z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
